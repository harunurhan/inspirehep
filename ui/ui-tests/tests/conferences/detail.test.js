const { routes, selectors } = require('../../utils/constants');
const { createPollyInstance } = require('../../utils/polly');
const { getMetaDescription } = require('../../utils/dom');
const {
  takeScreenShotForDesktop,
  takeScreenShotForMobile,
} = require('../../utils/screenshot');

describe('Conference Detail', () => {
  let polly;

  beforeEach(async () => {
    polly = await createPollyInstance('ConferenceDetail');

    await page.goto(routes.public.conferenceDetail1339293, {
      waitUntil: 'networkidle0',
    });
  });

  it('should match image snapshot for a Conference', async () => {
    await page.waitFor(selectors.searchResults);

    const desktopSS = await takeScreenShotForDesktop(page);
    expect(desktopSS).toMatchImageSnapshot();

    const mobileSS = await takeScreenShotForMobile(page);
    expect(mobileSS).toMatchImageSnapshot();
  });

  it('sets conference name as document title', async () => {
    const documentTitle = await page.title();

    expect(documentTitle).toMatch(
      /^4th International Conference on Micro Pattern Gaseous Detectors/
    );
  });

  it('sets conference short description as meta description', async () => {
    const metaDescription = await getMetaDescription(page);
    expect(metaDescription).toMatchSnapshot();
  });

  afterEach(async () => {
    await polly.stop();
  });
});
