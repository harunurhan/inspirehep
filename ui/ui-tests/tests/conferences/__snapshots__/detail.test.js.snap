// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Conference Detail sets conference short description as meta description 1`] = `"The modern photolithographic technology on flexible and standard PCB supports has allowed the invention and development of novel and very robust Micro-Patter..."`;
